rubicon/objc/objc.py

```diff
@@ -1324,6 +1325,10 @@ def __new__(cls, object_ptr):
         class_ptr = c_void_p(objc.object_getClass(object_ptr))
         objc_instance.__dict__['objc_class'] = ObjCClass(class_ptr)
 
+        # If it's an array, we need the internal counter in the instance to implement iterable.
```

Comment:
No you don't - because that makes the iterable object stateful. A ListIterator is a separate object from a list, so you can perform 2 parallel iterations over the same list.